src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -1,39 +1,84 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './NewCommentForm.scss';
+import { nanoid } from 'nanoid';
+import { Comment } from '../../types/comment';
 
-export const NewCommentForm: React.FC = () => (
-  <form className="NewCommentForm">
-    <div className="form-field">
-      <input
-        type="text"
-        name="name"
-        placeholder="Your name"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <div className="form-field">
-      <input
-        type="text"
-        name="email"
-        placeholder="Your email"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <div className="form-field">
-      <textarea
-        name="body"
-        placeholder="Type comment here"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <button
-      type="submit"
-      className="NewCommentForm__submit-button button"
-    >
-      Add a comment
-    </button>
-  </form>
-);
+type Props = {
+  postId: number;
+  addNewComment: (newComment: Comment) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({ postId, addNewComment }) => {
+  const [newName, setNewName] = useState<string>('');
+  const [newEmail, setNewEmail] = useState<string>('');
+  const [newBody, setNewBody] = useState<string>('');
+
+  const newComment: Comment = {
+    postId,
+    id: +nanoid(),
+    name: newName,
+    email: newEmail,
+    body: newBody,
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    addNewComment(newComment);
+    setNewName('');
+    setNewEmail('');
+    setNewBody('');
+  };
+
+  return (
+    <form className="NewCommentForm" onSubmit={(e) => handleSubmit(e)}>
+      <div className="form-field">
+        <input
+          required
+          type="text"
+          name="name"
+          value={newName}
+          placeholder="Your name"
+          className="NewCommentForm__input"
+          onChange={(event) => {
+            setNewName(event.target.value);
+          }}
```

Comment:
it's better to access target.value in method and just pass callback like:
```suggestion
          onChange={setNewName}
```